packages/synapse-react-client/src/components/HasAccess/HasAccessV2.tsx

```diff
@@ -302,15 +329,14 @@ export function HasAccessV2(props: HasAccessProps) {
         }}
       >
         <AccessIcon downloadType={fileHandleDownloadType} />
-      </button>
+      </Button>
     ) : (
-      <AccessIcon downloadType={fileHandleDownloadType} />
+      <AccessIcon downloadType={fileHandleDownloadType!} />
     )
 
   return (
     <span style={{ whiteSpace: 'nowrap' }}>
-      {iconContainer}
-      {accessRequirementsJsx}
+      {accessRequirementsJsx ? accessRequirementsJsx : iconContainer}
```

Comment:
So it looks like we are rendering the `iconContainer` when `accessRequirementsJsx` is null. Would it make more sense/be possible to this? 
1. Define `iconContainer` first
2. Return `iconContainer` instead of `null` in the `accessRequirementsJsx` logic

Then we could remove this ternary and it might be a bit more straightforward.

Reply:
Great idea, I'll make that improvement!